src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -224,7 +224,7 @@ private BoundExpression BindCompoundAssignment(AssignmentExpressionSyntax node,
                 if (!rightToLeftConversion.IsImplicit || !rightToLeftConversion.IsValid)
                 {
                     hasError = true;
-                    GenerateImplicitConversionError(diagnostics, node, rightToLeftConversion, right, leftType);
+                    GenerateImplicitConversionError(diagnostics, right.Syntax, rightToLeftConversion, right, leftType);
```

Comment:
this matches the logic of normal assignment conversion errors.  e.g.:

![image](https://user-images.githubusercontent.com/4564579/198372521-0d25d8b9-8e42-4258-b681-cb0bec818f97.png)

Looks like this may have just been an oversight.